core/domain/story_domain.py

```diff
@@ -0,0 +1,369 @@
+
+        Returns:
+            StoryNode. The corresponding StoryNode domain object.
+        """
+        node = cls(
+            node_dict['id'], node_dict['destination_node_ids'],
+            node_dict['acquired_skill_ids'],
+            node_dict['prerequisite_skill_ids'], node_dict['outline'],
+            node_dict['exploration_id'])
+
+        return node
+
+    @classmethod
+    def create_default_story_node(cls, node_id):
+        """Returns a StoryNode domain object with default values.
+
+        Args:
+            node_id: str. The id of the node.
+
+        Returns:
+            StoryNode. The StoryNode domain object with default
+            value.
+        """
+        return cls(node_id, [], [], [], '', '')
+
+
+class Story(object):
+    """Domain object for an Oppia Story."""
+
+    def __init__(
+            self, story_id, title, description, notes,
+            story_nodes, schema_version, language_code, version,
+            created_on=None, last_updated=None):
+        """Constructs a Question domain object.
+
+        Args:
+            story_id: str. The unique ID of the story.
+            title: str. The title of the story.
+            description: str. The high level desscription of the story.
+            notes: str. A set of notes, that describe the characters,
+                main storyline, and setting.
+            story_nodes: list(StoryNode). List of the story nodes present in
+                the story.
+            created_on: datetime.datetime. Date and time when the story is
+                created.
+            last_updated: datetime.datetime. Date and time when the
+                story was last updated.
+            schema_version: int. The schema version for the story nodes object.
+            language_code: str. The ISO 639-1 code for the language this
+                story is written in.
+            version: int. The version of the story.
+        """
+        self.id = story_id
+        self.title = title
+        self.description = description
+        self.notes = notes
+        self.story_nodes = story_nodes
+        self.schema_version = schema_version
+        self.language_code = language_code
+        self.created_on = created_on
+        self.last_updated = last_updated
+        self.version = version
+
+    def to_dict(self):
+        """Returns a dict representing this Story domain object.
+
+        Returns:
+            A dict, mapping all fields of Story instance.
+        """
+        return {
+            'id': self.id,
+            'title': self.title,
+            'description': self.description,
+            'notes': self.notes,
+            'language_code': self.language_code,
+            'schema_version': self.schema_version,
+            'version': self.version,
+            'story_nodes': [
+                node.to_dict() for node in self.story_nodes
+            ]
+        }
+
+    @classmethod
+    def from_dict(
```

Comment:
Btw, I'm wondering if we need this method at all. When will it be used?

If we want to keep it then maybe renaming it to something like create_new_from_dict() might be better, though in that case I wonder why we need to pass in created_on and last_updated.

(I know some of this is lifted from Exploration, but that has a lot of legacy stuff. So I just want to understand how it applies to the context of Story.)

Reply:
Currently its not used anywhere else. I'll remove it for now and add it later if it becomes necessary.